app/assets/javascripts/discourse/app/components/post/menu.gjs

```diff
@@ -0,0 +1,557 @@
+]);
+
+function smallUserAttributes(user) {
+  return {
+    template: user.avatar_template,
+    username: user.username,
+    post_url: user.post_url,
+    url: userPath(user.username_lower),
+    unknown: user.unknown,
+  };
+}
+
+export default class PostMenu extends Component {
+  @service appEvents;
+  @service capabilities;
+  @service currentUser;
+  @service keyValueStore;
+  @service modal;
+  @service menu;
+  @service site;
+  @service siteSettings;
+  @service store;
+
+  @tracked collapsed = true; // TODO some plugins will need a value transformer
+  @tracked isWhoLikedVisible = false;
+  @tracked likedUsers = [];
+  @tracked totalLikedUsers;
+  @tracked isWhoReadVisible = false;
+  @tracked readers = [];
+  @tracked totalReaders;
+
+  get buttonActions() {
+    return {
+      copyLink: this.args.copyLink,
+      deletePost: this.args.deletePost,
+      editPost: this.args.editPost,
+      toggleLike: this.toggleLike,
+      openAdminMenu: this.openAdminMenu,
+      recoverPost: this.args.recoverPost,
+      replyToPost: this.args.replyToPost,
+      setCollapsed: (value) => (this.collapsed = value),
+      share: this.args.share,
+      showDeleteTopicModal: this.showDeleteTopicModal,
+      showFlags: this.args.showFlags,
+      showMoreActions: this.showMoreActions,
+      toggleReplies: this.args.toggleReplies,
+      toggleWhoLiked: this.toggleWhoLiked,
+      toggleWhoRead: this.toggleWhoRead,
+    };
+  }
+
+  @cached
+  get staticMethodsContext() {
+    return Object.freeze({
+      canCreatePost: this.args.canCreatePost,
+      collapsed: this.collapsed,
+      currentUser: this.currentUser,
+      filteredRepliesView: this.args.filteredRepliesView,
+      isWhoLikedVisible: this.isWhoLikedVisible,
+      isWhoReadVisible: this.isWhoReadVisible,
+      isWikiMode: this.isWikiMode,
+      repliesShown: this.args.repliesShown,
+      replyDirectlyBelow:
+        this.args.nextPost?.reply_to_post_number ===
+          this.args.post.post_number &&
+        this.args.post.post_number !== this.args.post.filteredRepliesPostNumber,
+      showReadIndicator: this.args.showReadIndicator,
+      suppressReplyDirectlyBelow:
+        this.siteSettings.suppress_reply_directly_below,
+    });
+  }
+
+  get staticMethodsArgs() {
+    return Object.freeze({
+      context: this.staticMethodsContext,
+      post: this.args.post,
+    });
+  }
+
+  get context() {
+    return Object.freeze({
+      ...this.staticMethodsContext,
+      collapsedButtons: this.renderableCollapsedButtons,
+    });
+  }
+
+  @cached
+  get registeredButtons() {
+    // the DAG is not resolved now, instead we just use the object for convenience to pass a nice DAG API to be used
+    // in the value transformer, and extract the data to be used later to resolve the DAG order
+    const buttonsRegistry = applyValueTransformer(
+      "post-menu-buttons",
+      // TODO auto-generate the position coordinates based on the order of the elements
```

Comment:
Do we need to resolve this before merging?

Reply:
It's resolved. The positions are now auto-generated based on the configured items.